### PR TITLE
gh-96471: Correct docs for queue shutdown raises

### PR DESCRIPTION
**Pull request by EpicWink**
* Remove "raises ShutDown" from `Queue.join()` and `Queue.task_done()` documentation (including doc-string)
* Clarify `Queue.shutdown()` behaviour (it doesn't raise itself, but causes other methods to raise)
* Add that `Queue.task_done()` is called by `Queue.shutdown(immediate=True)` for each item in the queue

Caused by: #104750 

* Issue: gh-96471

----
📚 Documentation preview 📚: https://cpython-previews--115838.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/queue.rst [hunk 1]
```diff
@@ -188,9 +188,7 @@ fully processed by daemon consumer threads.
    that had been :meth:`put` into the queue).
 
    Raises a :exc:`ValueError` if called more times than there were items placed in
-   the queue.
-
-   Raises :exc:`ShutDown` if the queue has been shut down immediately.
+   the queue, or after an immediate shutdown.
```
Comment (gvanrossum) [inline Doc/library/queue.rst]: That's not entirely correct IIRC. Even after an immediate shutdown, if there are items still being processed, calling `task_done()` for those is allowed. Basically once a `get()` has returned an item, you can call `task_done()` for it (once) any time. Only `get()` and `put()` check the `self.is_shutdown` variable.

File: Lib/queue.py [hunk 2]
```diff
@@ -227,13 +223,13 @@ def get_nowait(self):
...
 
         By default, gets will only raise once the queue is empty. Set
         'immediate' to True to make gets raise immediately instead.
 
         All blocked callers of put() will be unblocked, and also get()
-        and join() if 'immediate'. The ShutDown exception is raised.
+        and join() if 'immediate'.
         '''
         with self.mutex:
             self.is_shutdown = True
```
Comment (gvanrossum) [inline Lib/queue.py:238]: A few lines below is an unused `n_items = self._qsize()` line -- can you remove that too?

**Review by gvanrossum — COMMENTED** [review]
Thanks! Two nits and then I'll merge this. (I feel bad that I didn't catch this in review.)

**Review by gvanrossum — COMMENTED** [review]
Please fix the nits I called out.

**Review by gvanrossum — CHANGES_REQUESTED** [review]
Please fix the nits I called out.

**Comment by EpicWink** [comment]
I have made the requested changes; please review again

File: Doc/library/queue.rst [hunk 3]
```diff
@@ -187,11 +187,12 @@ fully processed by daemon consumer threads.
    processed (meaning that a :meth:`task_done` call was received for every item
    that had been :meth:`put` into the queue).
 
+   :meth:`shutdown(immediate=True)` calls :meth:`task_done` for each remaining item
```
Comment (gvanrossum) [inline Doc/library/queue.rst]:
```suggestion
   ``shutdown(immediate=True)`` calls :meth:`task_done` for each remaining item
```

**Review by gvanrossum — APPROVED** [review]

**Pull request merged**